apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/UserInfoController.java

```diff
@@ -53,13 +53,17 @@ public UserInfoController(
     this.userService = userService;
   }
 
-
   @PreAuthorize(value = "@permissionValidator.isSuperAdmin()")
   @PostMapping("/users")
   public void createOrUpdateUser(@RequestBody UserPO user) {
     if (StringUtils.isContainEmpty(user.getUsername(), user.getPassword())) {
       throw new BadRequestException("Username and password can not be empty.");
     }
+    if (UserPasswordChecker.isWeakPassword(user.getPassword())) {
+      throw new BadRequestException(
+          "Password needs a number and lowercase letter and between 8~20 characters. "
+              + "And cannot be a weak password.");
```

Comment:
I think you should adapt the message. Makes it more useable and clear to the user what the real reason is. 
It can be not matching to the pattern, `weak password` or both.
If it is just one and you display both reasons in the error message the user might not understand it why you get both. 

PS: you should also explain what a `weak password` is in the exception

Reply:
Ok, I've split the error message, and wrap it with `CheckResult`.